src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Phases/PhaseForm.razor

```diff
@@ -1,41 +1,40 @@
 ﻿@using NTS.Domain.Setup.Entities;
 @using NTS.Judge.Setup.Competitions
 @using Not.Blazor.Forms;
+@using Not.Blazor.TM;
+@using Not.Blazor.TM.Models;
 @using Not.Blazor.TM.Forms;
 @using Not.Blazor.TM.Forms.Components;
 
 @inherits NotForm<Phase>;
 
 <EditForm Model="_model">
-    <MudSelect @ref="_lapField" @bind-Value="_model.SelectedLap" Label="@Localizer.Get("Phase Distance")" @onchange="(_) => SubmitUpdate()"
-        Placeholder="@Localizer.Get("Pick lap length")">
-        @foreach(var item in Items)
-        {
-            <MudSelectItem Value="item">@item</MudSelectItem>
-        }
-    </MudSelect>
+    <NotSelect T="Loop" Label="@Localizer.Get("Loop distance")" Placeholder="@Localizer.Get("Select a loop")"
+               Items="_items" @bind-Value="_model.Loop" />
     <MudNumericField T="int" @ref="_recoveryField" @bind-Value="_model.Recovery" Min="0" Label="@Localizer.Get("Recovery Time")" />
     <MudNumericField T="int" @ref="_restField" @bind-Value="_model.Rest" Min="0" Label="@Localizer.Get("Rest Duration")" />
 </EditForm>
 
 @code {
-    private MudSelect<Lap> _lapField = default!;
+    private NotSelect<Loop> _loopField = default!;
     private MudNumericField<int> _recoveryField = default!;
     private MudNumericField<int> _restField = default!;
     private PhaseFormModel _model { get; set; } = new();
-    private IEnumerable<Lap> Items { get; set; } = default!;
+
+    private IEnumerable<NotListModel<Loop>> _items = default!;
```

Comment:
Isn't `_items` a bit generic? What happens if we add a second `NotSelect` for example?